File: hw/medina/u1/hw14SafeInJsonChickenFarmSimulator/java/src/model/Coop.java
package model;

import java.util.ArrayList;
import java.util.List;

public class Coop {
    private int id;
    private String location;
    private List<Chicken> chickens;

    public Coop(int id, String location) {
        this.id = id;
        this.location = location;
        this.chickens = new ArrayList<>();
    }

    public int getId() { return id; }
    public String getLocation() { return location; }
    public List<Chicken> getChickens() { return chickens; }

    public void addChicken(Chicken chicken) { chickens.add(chicken); }
    public void removeChicken(int id) { chickens.removeIf(c -> c.getId() == id); }

    public Chicken findChickenById(int id) {
        for (Chicken c : chickens) if (c.getId() == id) return c;
        return null;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Coop ID: ").append(id).append(" | Location: ").append(location)
          .append(" | Chickens: ").append(chickens.size()).append("\n");
        for (Chicken c : chickens) sb.append("  ").append(c.toString()).append("\n");
        return sb.toString();
    }
}

File: hw/medina/u1/hw14SafeInJsonChickenFarmSimulator/python/model/coop.py
from model.chicken import Chicken

class Coop:
    def __init__(self, id, location):
        self.id = id
        self.location = location
        self.chickens = []

    def add_chicken(self, chicken):
        self.chickens.append(chicken)

    def remove_chicken(self, chicken_id):
        self.chickens = [c for c in self.chickens if c.id != chicken_id]

    def get_chicken_by_id(self, chicken_id):
        for chicken in self.chickens:
            if chicken.id == chicken_id:
                return chicken
        return None

    def to_dict(self):
        return {
            "id": self.id,
            "location": self.location,
            "chickens": [chicken.to_dict() for chicken in self.chickens]
        }

    @staticmethod
    def from_dict(data):
        coop = Coop(data["id"], data["location"])
        coop.chickens = [Chicken.from_dict(ch) for ch in data.get("chickens", [])]
        return coop

    def __str__(self):
        result = f"\n {self.location} (ID: {self.id}) contains:\n"
        if not self.chickens:
            result += " - (Empty)\n"
        else:
            for chicken in self.chickens:
                result += f" - {chicken.name} (ID: {chicken.id})\n"
        result += "------------------------------------"
        return result
